From 12f78653080fb2bed3f2eb60559b278bb68f92b8 Mon Sep 17 00:00:00 2001
From: Rico Ganahl <rico.ganahl@bytesatwork.ch>
Date: Wed, 31 Jan 2024 10:55:12 +0100
Subject: [PATCH] bytesatwork: bawconfig: introduce nxp imx8mp

Add hardware revision of bytes at work NXP IMX 8M Plus SoM.

Signed-off-by: Rico Ganahl <rico.ganahl@bytesatwork.ch>
(cherry picked from commit 97c39a73a1262cc66bc98553404a98b1e6e24acd)
---
 board/bytesatwork/baw_config/baw_config.c | 7 +++++++
 board/bytesatwork/baw_config/baw_config.h | 7 +++++++
 2 files changed, 14 insertions(+)

diff --git a/board/bytesatwork/baw_config/baw_config.c b/board/bytesatwork/baw_config/baw_config.c
index 73a59952e6..70b2ef00cb 100644
--- a/board/bytesatwork/baw_config/baw_config.c
+++ b/board/bytesatwork/baw_config/baw_config.c
@@ -18,6 +18,7 @@ const struct baw_config_name_pair baw_config_pcb_name[] = {
 	{ M8_PCB_REV_0_1, "M8 Rev 0.1" },
 	{ M8_PCB_REV_1_0, "M8 Rev 1.0" },
 	{ M8_PCB_REV_1_1, "M8 Rev 1.1" },
+	{ M7_PCB_REV_1_0, "M7 Rev 1.0" },
 	{ -1, NULL },
 };
 
@@ -40,6 +41,12 @@ const struct baw_config_name_pair baw_config_ram_name[] = {
 	{ M8_RAM_MT53E4G16D4NQ_046, "MT53E4G16D4NQ-046 8 GB LPDDR4 @ 1600 MHz"},
 	{ M5_RAM_K4B4G1646DBIK0,   "K4B4G1646DBIK0 512 MB DDR3 @ 533MHz" },
 	{ M5_RAM_MT41K512M16HA107, "MT41K512M16HA107 1024 MB DDR3 @ 533MHz" },
+	{ M7_RAM_MT53E128M32D2DS_053, "MT53E128M32D2DS-053 512 MB LPDDR4 @ 2000 MHz" },
+	{ M7_RAM_MT53E256M32D2DS_053, "MT53E256M32D2DS-053 1 GB LPDDR4 @ 2000 MHz" },
+	{ M7_RAM_MT53E384M32D2DS_053, "MT53E384M32D2DS-053 1.5 GB LPDDR4 @ 2000 MHz" },
+	{ M7_RAM_MT53D512M32D2DS_053, "MT53D512M32D2DS-053 2 GB LPDDR4 @ 2000 MHz" },
+	{ M7_RAM_MT53E768M32D4DT_053, "MT53E768M32D4DT-053 3 GB LPDDR4 @ 2000 MHz" },
+	{ M7_RAM_MT53E1G32D2FW_046, "MT53E1G32D2FW-046 4 GB LPDDR4 @ 2000 MHz" },
 	{ -1, NULL },
 };
 
diff --git a/board/bytesatwork/baw_config/baw_config.h b/board/bytesatwork/baw_config/baw_config.h
index 9e08586610..df7c7e1cd6 100644
--- a/board/bytesatwork/baw_config/baw_config.h
+++ b/board/bytesatwork/baw_config/baw_config.h
@@ -22,6 +22,7 @@ enum baw_config_pcb {
 	M8_PCB_REV_0_1 = 9,
 	M8_PCB_REV_1_0 = 10,
 	M8_PCB_REV_1_1 = 11,
+	M7_PCB_REV_1_0 = 12,
 };
 
 enum baw_config_ram {
@@ -46,6 +47,12 @@ enum baw_config_ram {
 	M8_RAM_MT53E4G16D4NQ_046 = 16,
 	M5_RAM_K4B4G1646DBIK0 = 17,
 	M5_RAM_MT41K512M16HA107 = 18,
+	M7_RAM_MT53E128M32D2DS_053 = 19,
+	M7_RAM_MT53E256M32D2DS_053 = 20,
+	M7_RAM_MT53E384M32D2DS_053 = 21,
+	M7_RAM_MT53D512M32D2DS_053 = 22,
+	M7_RAM_MT53E768M32D4DT_053 = 23,
+	M7_RAM_MT53E1G32D2FW_046 = 24,
 };
 
 enum baw_config_flash {
-- 
2.39.2

